Model/FileIndex.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace SomethingDigital\UpgradeHelper\Model;
+
+class FileIndex
+{
+    const THEME_OVERRIDE = 'theme_override';
+
+    const MODULE_OVERRIDE = 'module_override';
+
+    const INTERESTING_EXTENSIONS = [
+            'phtml',
+            'js',
+            'html',
+            'less'
+    ];
+
+    const WHITELISTED_BASENAMES = [
+        'requirejs-config.js'
+    ];
+
+    /**
+     * @var array[]
+     */
+    private $index;
```

Comment:
Can we elaborate on what will be stored in this variable in a comment here?

Can we demonstrate the stored data structure.